foundations/foundation-vertx/src/main/java/org/apache/servicecomb/foundation/vertx/metrics/DefaultHttpClientMetrics.java

```diff
@@ -77,8 +83,13 @@ public void dequeueRequest(DefaultClientEndpointMetric endpointMetric, Object ta
 
   @Override
   public void endpointConnected(DefaultClientEndpointMetric endpointMetric, DefaultHttpSocketMetric socketMetric) {
-    socketMetric.setEndpointMetric(endpointMetric);
-    endpointMetric.onConnect();
+    //only http1.1 will invoke this method, just make a check
+    if (endpointMetric != null) {
```

Comment:
it's vertx bug: https://github.com/eclipse-vertx/vert.x/issues/2660
please add comments, and remove the hack code after vertx fix the bug.

Reply:
ok, I see